codersdk/inboxnotification.go

```diff
@@ -0,0 +1,123 @@
+	Title      string                    `json:"title"`
+	Content    string                    `json:"content"`
+	Icon       string                    `json:"icon"`
+	Actions    []InboxNotificationAction `json:"actions"`
+	ReadAt     *time.Time                `json:"read_at"`
+	CreatedAt  time.Time                 `json:"created_at" format:"date-time"`
+}
+
+type InboxNotificationAction struct {
+	Label string `json:"label"`
+	URL   string `json:"url"`
+}
+
+type GetInboxNotificationResponse struct {
+	Notification InboxNotification `json:"notification"`
+	UnreadCount  int               `json:"unread_count"`
+}
+
+func uuidSliceToString(s []uuid.UUID) string {
+	resp := ""
+	for idx, v := range s {
+		resp += v.String()
+		if idx < len(s)-1 {
+			resp += ","
+		}
+	}
+
+	return resp
+}
+
+type ListInboxNotificationsRequest struct {
+	Targets        []uuid.UUID
+	Templates      []uuid.UUID
+	ReadStatus     string
+	StartingBefore uuid.UUID
+}
+
+type ListInboxNotificationsResponse struct {
+	Notifications []InboxNotification `json:"notifications"`
+	UnreadCount   int                 `json:"unread_count"`
+}
+
+func ListInboxNotificationsRequestToQueryParams(req ListInboxNotificationsRequest) []RequestOption {
+	var opts []RequestOption
+	if len(req.Targets) > 0 {
+		opts = append(opts, WithQueryParam("targets", uuidSliceToString(req.Targets)))
+	}
+	if len(req.Templates) > 0 {
+		opts = append(opts, WithQueryParam("templates", uuidSliceToString(req.Templates)))
+	}
+	if req.ReadStatus != "" {
+		opts = append(opts, WithQueryParam("read_status", req.ReadStatus))
+	}
+	if req.StartingBefore != uuid.Nil {
+		opts = append(opts, WithQueryParam("starting_before", req.StartingBefore.String()))
+	}
+
+	return opts
+}
+
+func (c *Client) ListInboxNotifications(ctx context.Context, req ListInboxNotificationsRequest) (ListInboxNotificationsResponse, error) {
+	res, err := c.Request(
+		ctx, http.MethodGet,
+		"/api/v2/notifications/inbox",
+		nil, ListInboxNotificationsRequestToQueryParams(req)...,
+	)
+	if err != nil {
+		return ListInboxNotificationsResponse{}, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return ListInboxNotificationsResponse{}, ReadBodyAsError(res)
+	}
+
+	var listInboxNotificationsResponse ListInboxNotificationsResponse
+	return listInboxNotificationsResponse, json.NewDecoder(res.Body).Decode(&listInboxNotificationsResponse)
+}
+
+type UpdateInboxNotificationReadStatusRequest struct {
+	IsRead bool `json:"is_read"`
+}
+
+type UpdateInboxNotificationReadStatusResponse struct {
+	Notification InboxNotification `json:"notification"`
+	UnreadCount  int               `json:"unread_count"`
+}
+
+func (c *Client) UpdateInboxNotificationReadStatus(ctx context.Context, notifID uuid.UUID, req UpdateInboxNotificationReadStatusRequest) (UpdateInboxNotificationReadStatusResponse, error) {
+	res, err := c.Request(
+		ctx, http.MethodPut,
+		fmt.Sprintf("/api/v2/notifications/inbox/%v/read-status", notifID),
```

Comment:
`%v` in the URL suggests, that it should be possible to access a single notification via REST endpoint `/api/v2/notifications/inbox/%v` which is not true.

Alternatively, we can change this to `/api/v2/notifications/inbox/update`: `{ "notification_ids": [ ... ], "action": "MARK_READ" }`.

PS. Ignore this if it is too late to change the API.